src/contracts/atlas/Escrow.sol

```diff
@@ -430,57 +421,144 @@ abstract contract Escrow is AtlETH {
+                // DAppControl is attacking solver contract - treat as AlteredControl
+                result = 1 << uint256(SolverOutcome.AlteredControl);
+            } else if (errorSwitch == CallbackNotCalled.selector) {
+                result = 1 << uint256(SolverOutcome.SolverOpReverted);
+            } else {
+                result = 1 << uint256(SolverOutcome.EVMError);
+            }
+        }
+    }
+
+    /// @notice Executes the SolverOperation logic, including preSolver and postSolver hooks via the Execution
+    /// Environment, as well as the actual solver call directly from Atlas to the solver contract.
+    /// @param bidAmount The bid amount associated with the SolverOperation.
+    /// @param gasLimit The gas limit for executing the SolverOperation.
+    /// @param key The EscrowKey struct containing lock data and the Execution Environment address.
+    /// @param solverOp The SolverOperation to be executed.
+    /// @param dAppReturnData Data returned from previous call phases.
+    /// @return solverTracker Additional data for handling the solver's bid in different scenarios.
+    function solverCall(
+        uint256 bidAmount,
+        uint256 gasLimit,
+        EscrowKey calldata key,
+        SolverOperation calldata solverOp,
+        bytes calldata dAppReturnData
+    )
+        external
+        payable
+        returns (SolverTracker memory solverTracker)
+    {
+        if (msg.sender != address(this)) revert InvalidEntry();
+
+        bool success;
+
+        // ------------------------------------- //
+        //             Pre-Solver Call           //
+        // ------------------------------------- //
+
         bytes memory data =
-            abi.encodeCall(IExecutionEnvironment.solverMetaTryCatch, (bidAmount, gasLimit, solverOp, dAppReturnData));
+            abi.encodeCall(IExecutionEnvironment.solverPreTryCatch, (bidAmount, solverOp, dAppReturnData));
+        (success, data) = key.executionEnvironment.call(abi.encodePacked(data, key.holdPreSolverLock().pack()));
 
-        data = abi.encodePacked(data, lockBytes);
+        // If ExecutionEnvironment.solverPreTryCatch() failed, bubble up the error
+        if (!success) {
+            assembly {
+                revert(add(data, 32), mload(data))
+            }
+        }
 
-        (success, data) = environment.call{ value: solverOp.value }(data);
+        // Update solverTracker with returned data
+        solverTracker = abi.decode(data, (SolverTracker));
 
-        if (success) {
-            return uint256(0);
+        // ------------------------------------- //
+        //              Solver Call              //
+        // ------------------------------------- //
+
+        // Set the solver lock - will perform accounting checks if value borrowed. If `_trySolverLock()` returns false,
+        // we revert here and catch the error in `_solverOpWrapper()` above
+        if (!_trySolverLock(solverOp)) revert SolverMetaTryCatchIncorrectValue();
```

Comment:
If `_trySolverLock()` fails in `solverCall()` we revert with `error SolverMetaTryCatchIncorrectValue()` and catch it in `_solverOpWrapper()` and set `uint result` to reflect the error type.

But there is no specific case in that "try-catch" that would match that error, so I think it would get assigned as `SolverOutcome.EVMError` which is interpreted as the solver's fault (solver owes full gas refund).

I think we'd want this error to be interpreted as solver's fault, but we should probably catch this specific revert message and use a new corresponding SolverOutcome value for debugging and simulator clarity.

Reply:
good catch.

Reply:
Going to use the "InsufficientEscrow" outcome and create an error for it.

Reply:
823aeda